kernel/api/src/main/java/org/sakaiproject/content/api/FileSystemHandler.java

```diff
@@ -0,0 +1,43 @@
+package org.sakaiproject.content.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This is the api for reading and writing files to some file system.
+ *
+ */
+public interface FileSystemHandler {
+
+    /**
+     * Retrieves an input stream from the file.
+     * 
+     * @param id The id of the resource. Will not be null or empty.
+     * @param root The root of the storage. Could be null or empty.
+     * @param filePath The path to the file. Will not be null or empty.
+     * @return The valid input stream. Must not be null.
+     * @throws IOException If the stream could not be created.
+     */
+    public InputStream getInputStream(String id, String root, String filePath) throws IOException;
+
+    /**
+     * Save the file from the input stream to the path and return the content size.
+     * 
+     * @param id The id of the resource. Will not be null or empty.
+     * @param root The root of the storage. Could be null or empty.
+     * @param filePath The path to save the file to. Will not be null or empty.
+     * @param stream The stream to read the file from.
```

Comment:
Yes, this seems right to me. Unless we want to do some sort of transactional callback interface, the caller will be responsible for cleaning up. This is probably expectable and reasonable.